all-in-one-apim/modules/integration-v2/tests-integration/cucumber-tests/CLAUDE.md

```
@@ -180,6 +203,32 @@ mvn install -pl tests-common/integration-test-utils -DskipTests
 
+**Rebuilding the APIM container image after a PRODUCT change (carbon-apimgt or distribution).** A product-code
+change does NOT reach a test run by installing the jar to `.m2` — the container runs the product from a
+**distribution zip baked into the image**, not from `.m2`. `mvn test` deliberately **reuses** the existing
+`wso2am:${apim.server.version}-jdk21` image, so it will keep running the OLD product until the image is rebuilt.
+The chain, and how to refresh it:
+1. **Rebuild the product zip with your change.** Install carbon-apimgt (`mvn install` there) so the fixed jar
+   is in `.m2`, then rebuild the distribution so `all-in-one-apim/modules/distribution/product/target/wso2am-${apim.server.version}.zip`
+   bundles it (verify: the zip's `repository/components/plugins/org.wso2.carbon.apimgt.gateway_<ver>.jar` is the
+   fixed one — OSGi bundles use an **underscore** before the version, not a hyphen).
+2. **Rebuild the image** from the integration-v2 root:
+   ```
```

markdownlint-cli2 (0.23.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)